lib/using_restricted_array.rb

```diff
@@ -6,56 +6,116 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: 0(n)
+# Space complexity: 0(1)
 def length(array)
-  raise NotImplementedError
+  # i is the array length
+  i = 0
+  while array[i] != nil
+    i+=1
+  end
+  i
```

Comment:
I just suggest putting in the return keyword for readability.

```suggestion
  return i
```